stdlib/public/runtime/CMakeLists.txt

```diff
@@ -103,16 +103,39 @@ if(SWIFT_BUILD_STATIC_STDLIB AND "${sdk}" STREQUAL "LINUX")
   string(TOLOWER "${sdk}" lowercase_sdk)
 
   # These two libraries are only used with the static swiftcore
-  add_swift_library(swiftImageInspectionShared STATIC
+  add_swift_library(swiftImageInspectionShared TARGET_LIBRARY STATIC
     ImageInspectionELF.cpp
     C_COMPILE_FLAGS ${swift_runtime_library_compile_flags}
-    LINK_FLAGS ${swift_runtime_linker_flags})
-  set_target_properties(swiftImageInspectionShared PROPERTIES
-    ARCHIVE_OUTPUT_DIRECTORY "${SWIFTSTATICLIB_DIR}/${lowercase_sdk}")
+    LINK_FLAGS ${swift_runtime_linker_flags}
+    INSTALL_IN_COMPONENT stdlib)
+
+  foreach(arch IN LISTS SWIFT_SDK_${sdk}_ARCHITECTURES)
+    set(FragileSupportLibrary swiftImageInspectionShared-${SWIFT_SDK_${sdk}_LIB_SUBDIR}-${arch})
+    set(LibraryLocation ${SWIFTSTATICLIB_DIR}/${lowercase_sdk}/${arch})
+    add_custom_command_target(swift_image_inspection_${arch}_static
+      COMMAND
+        "${CMAKE_COMMAND}" -E copy $<TARGET_FILE:${FragileSupportLibrary}> ${LibraryLocation}
+      OUTPUT
+         "${LibraryLocation}/libswiftImageInspectionShared.a"
+      DEPENDS
+         ${FragileSupportLibrary})
+    swift_install_in_component(stdlib
+      FILES $<TARGET_FILE:swift_image_inspection_${arch}_static>
+      DESTINATION "lib/swift_static/${lowercase_sdk}/${arch}")
+  endforeach()
 
-  swift_install_in_component(stdlib
-    TARGETS swiftImageInspectionShared
-    DESTINATION "lib/swift_static/${lowercase_sdk}")
+  set(FragileSupportLibraryPrimary swiftImageInspectionShared-${SWIFT_SDK_${sdk}_LIB_SUBDIR}-${SWIFT_PRIMARY_VARIANT_ARCH})
+  set(LibraryLocationPrimary ${SWIFTSTATICLIB_DIR}/${lowercase_sdk})
+  add_custom_command_target(swift_image_inspection_static_primary_arch
+      COMMAND
+         "${CMAKE_COMMAND}" -E copy $<TARGET_FILE:${FragileSupportLibraryPrimary}> ${LibraryLocationPrimary}
+      OUTPUT
+         "${LibraryLocationPrimary}/libswiftImageInspectionShared.a"
+      DEPENDS
+         ${FragileSupportLibraryPrimary})
+    swift_install_in_component(stdlib
+      FILES $<TARGET_FILE:swift_image_inspection_static_primary_arch>
+      DESTINATION "lib/swift_static/${lowercase_sdk}")
```

Comment:
Shouldn't the previous for loop cover this already?

Reply:
This is a temporary addition to put it at `lib/swift_static/linux/libswiftImageInspectionShared.a` since the compiler adds this as the linker path on master. The above loop puts them at `lib/swift_static/linux/x86_64`.

One of the PRs on top of this removes the code that does things for non-arch specific paths.